tracer/src/Datadog.Trace/Configuration/ConfigurationSources/Telemetry/ConfigurationBuilder.cs

```diff
@@ -115,6 +122,78 @@ public string AsString(string defaultValue, Func<string, bool>? validator)
+                      ?? (FallbackKey2 is null ? null : Source.GetString(FallbackKey2, Telemetry, validator, recordValue))
+                      ?? (FallbackKey3 is null ? null : Source.GetString(FallbackKey3, Telemetry, validator, recordValue));
+
+            string? returnValue = null;
+            bool datadogConfigurationIsPresent = false;
+            if (result is { Result: { } value, IsValid: { } resultIsValid })
+            {
+                datadogConfigurationIsPresent = true;
+                if (resultIsValid)
+                {
+                    returnValue = value;
+                }
+            }
+            else
+            {
+                datadogConfigurationIsPresent = Source.IsPresent(Key)
+                                             || (FallbackKey1 is null ? false : Source.IsPresent(FallbackKey1))
+                                             || (FallbackKey2 is null ? false : Source.IsPresent(FallbackKey2))
+                                             || (FallbackKey3 is null ? false : Source.IsPresent(FallbackKey3));
+            }
+
+            // If there's a Datadog configuration present, check if a corresponding OpenTelemetry key is present so we can log the conflicting keys
+            if (datadogConfigurationIsPresent && Source.IsPresent(openTelemetryKey))
+            {
+                Log.Warning(
+                    "Both Datadog configuration {DatadogConfiguration} and OpenTelemetry configuration {OpenTelemetryConfiguration} are set. The Datadog configuration will be used.",
+                    Key,
+                    openTelemetryKey);
+                OpenTelemetryHelpers.GetConfigurationMetricTags(openTelemetryKey, out var openTelemetryConfig, out var datadogConfig);
+                TelemetryFactory.Metrics.RecordCountOpenTelemetryConfigHiddenByDatadogConfig(datadogConfig, openTelemetryConfig);
+            }
+            else if (Source.IsPresent(openTelemetryKey))
+            {
+                var openTelemetryResult = openTelemetryConverter switch
+                {
+                    null => Source.GetString(openTelemetryKey, Telemetry, validator, recordValue),
+                    _ => Source.GetAs(openTelemetryKey, Telemetry, openTelemetryConverter, validator, recordValue),
+                };
+
+                if (openTelemetryResult is { Result: { } openTelemetryValue, IsValid: true })
+                {
+                    return openTelemetryValue;
+                }
+                else
+                {
+                    Log.Warning("OpenTelemetry configuration {OpenTelemetryConfiguration} is invalid.", openTelemetryKey);
+                    OpenTelemetryHelpers.GetConfigurationMetricTags(openTelemetryKey, out var openTelemetryConfig, out var datadogConfig);
+                    TelemetryFactory.Metrics.RecordCountOpenTelemetryConfigInvalid(datadogConfig, openTelemetryConfig);
```

Comment:
I think if we're going to do this we should pass in `IMetricsTelemetryCollector` in the same way we do the `IConfigurationTelemetry`, instead of using the global instance. It could be problematic for testing and/or multiple-initialization issues otherwise I think.

Reply:
Ack, in e8fc0b590dbb733caa248882d8c6215cdd9a4a57 I've removed calls to the static `TelemetryFactory.Metrics`. We can try that for a follow-up implementation of the telemetry metrics signals